klippy/extras/dockable_probe.py

```diff
@@ -0,0 +1,596 @@
+                                 " states are UNKNOWN, ATTACHED, and DOCKED")
+    def cmd_GET_PROBE_STATUS(self, gcmd):
+        s = self.get_probe_state()
+        state = next(v for k, v
+                     in enumerate(['UNKNOWN', 'ATTACHED', 'DOCKED'])
+                     if s == k)
+        gcmd.respond_info('Probe Status: %s' % (state))
+    def _handle_ready(self):
+        self.last_verify_time = 0
+        self.last_verify_state = PROBE_UNKNOWN
+    def get_probe_state(self):
+        curtime = self.printer.get_reactor().monotonic()
+        last = self.last_verify_time
+        if (self.last_verify_state == PROBE_UNKNOWN
+            or curtime > self.last_verify_time + PROBE_VERIFY_DELAY):
+            self.last_verify_time = curtime
+            a = self.probe_sense_pin(curtime)
+            d = self.dock_sense_pin(curtime)
+            if a and not d:
+                self.last_verify_state = PROBE_ATTACHED
+            elif d and not a:
+                self.last_verify_state = PROBE_DOCKED
+            else:
+                self.last_verify_state = PROBE_UNKNOWN
+        return self.last_verify_state
+
+class DockableProbe:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.gcode = self.printer.lookup_object('gcode')
+        self.name = config.get_name()
+
+        # Configuration Options
+        self.position_endstop = config.getfloat('z_offset')
+        self.x_offset = config.getfloat('x_offset', 0.)
+        self.y_offset = config.getfloat('y_offset', 0.)
+        self.speed = config.getfloat('speed', 5.0, above=0.)
+        self.lift_speed = config.getfloat('lift_speed', self.speed, above=0.)
+        self.dock_retries = config.getint('dock_retries', 0) + 1
+        self.travel_speed = config.getfloat('travel_speed',
+                                             self.speed, above=0.)
+        self.attach_speed = config.getfloat('attach_speed',
+                                             self.travel_speed, above=0.)
+        self.detach_speed = config.getfloat('detach_speed',
+                                             self.travel_speed, above=0.)
+        self.dock_fixed_z = config.getboolean('dock_fixed_z', False)
+        self.allow_delayed_detach = config.getboolean('allow_delayed_detach',
+                                                       False)
+        self.sample_retract_dist = config.getfloat('sample_retract_dist', 2.,
+                                                   above=0.)
+
+        def parseCoord(name, c=3):
+            try:
+                p = [None] * 3
+                val = config.get(name, None)
+                if not val:
+                    return None
+                p[0:c] = map(lambda x: float(x.strip()),
+                             val.split(','))[0:c]
+            except:
+                e = "Unable to parse {0} in {1}"
+                raise config.error(e.format(self.name, name))
```

Comment:
Arguments `{0}` (self.name) and `{1}` (name) are swapped. Currently, this outputs the following into the log:
```
configparser.Error: Unable to parse dockable_probe in approach_position
```

Reply:
I have not tested this on python 3 yet but that should be easy enough to resolve.

Reply:
Yeah, easy to fix. Here what I did for my personal Klipper branch: https://github.com/wlhlm/klipper/commit/11e60b64b7ee5ecb5c8e6c9c8f06b33fe87f369a

Other than that it's working fine.